webpack/assets/javascripts/react_app/components/Pagination/PaginationWrapper.js

```diff
@@ -42,6 +43,7 @@ const PaginationWrapper = props => {
       dropdownButtonId={dropdownButtonId}
       pagination={pagination}
       {...otherProps}
+      messages={translateObject(Paginator.defaultProps.messages)}
```

Comment:
messages are already given [here](https://github.com/theforeman/foreman/blob/5345cbdb0f6862b19a8293f9e9c4229523f4026f/webpack/assets/javascripts/react_app/components/Pagination/Pagination.js#L53),  is it overrode?

Reply:
```
PaginationWrapper.defaultProps ={...
  ...Paginator.defaultProps,	
...}
```
`Paginator.defaultProps.messages ` has the untranslated text so the messages prop by default is the untranslated text .
In the pagination component it uses ` {...otherProps}`  which contains the default messages prop  so it overrides the translated messages in the line you sent

Reply:
how about overriding the default `messages` prop only once?
```js
// /components/Pagination/Pagination.js
  const messages = {
    firstPage: __('First Page'),
    previousPage: __('Previous Page'),
    currentPage: __('Current Page'),
    nextPage: __('Next Page'),
    lastPage: __('Last Page'),
    perPage: __('per page'),
    of: __('of'),
  };
  return (
    <Paginator
      pagination={pageOpts}
      // props
      {...otherProps}
      messages={messages}
    />
```
With this, the `PaginationWrapper` still gets the translated messages which override the default prop.

Reply:
I can, but as I stated in the PR 
Didn't want to override the custom messages prop in Pagination as it is the more "flexible" component.
Do we want to allow flexible messages or should we always override them in `Pagination.js`?

Reply:
We need to have the strings themselves somewhere in foreman for them to be extracted for translations, right now iiuc the strings are only present in patternfly which isn't extracted.

Reply:
Thats a different issue :) 
But I'll fix it here as well and always override the messages.
Thanks!

Reply:
We can still keep the flexibility by omitting the `messages` default prop from `PaginationWrapper`

Reply:
Thanks :+1: 
though the component is still not flexible,
how about adding the ability to override the translated default messages, something like this:
```js
  return (
    <Paginator
      pagination={pageOpts}
      // props
      {...otherProps}
      messages={messages || defaultMessage} // messages comes from the consumers
    />
```

Reply:
The messages don't override the given props
```
 messages={messages}
 {...otherProps}
```

Reply:
Do we actually ever need to override the messages?

Reply:
These override messages prop was already given to consumers, changing that is a breaking changes.